core/new-gui/src/app/workspace/component/workspace.component.ts

```diff
@@ -162,11 +169,17 @@ export class WorkspaceComponent implements OnInit {
         this.workflowCacheService.setCacheWorkflow(this.workflowActionService.getWorkflow());
       },
       () => {
-        // TODO: replace with a proper error message with the framework
-        alert('You don\'t have access to this workflow, please log in with another account');
         this.workflowCacheService.resetCachedWorkflow();
+        this.workflowActionService.reloadWorkflow(this.workflowCacheService.getCachedWorkflow());
+        this.noAccessToWorkflow();
+
       }
     );
   }
 
+  private noAccessToWorkflow(): void {
+    this.location.go(`/`);
+    // TODO: replace with a proper error message with the framework
+    alert('You don\'t have access to this workflow, please log in with another account');
```

Comment:
if there's no access, do we need to clear the workflow data? I don't see the code that clears the workflow. I saw we are still reloading the cached workflow. Or does location.go() solves it?

Reply:
no. `location.go()` only changes url in address bar (location). it does not load any components. The clear workflow code was outside this method, now I moved them inside.